src/MorphoScripts.sol

```diff
@@ -0,0 +1,182 @@
+                // Repay max
+                IERC20(marketParams.loanToken).forceApprove(morpho, type(uint256).max);
+                IMorpho(morpho).repay({
+                    marketParams: marketParams,
+                    assets: 0,
+                    shares: IMorpho(morpho).position(marketId(marketParams), address(this)).borrowShares,
+                    onBehalf: address(this),
+                    data: new bytes(0)
+                });
+                IERC20(marketParams.loanToken).forceApprove(morpho, 0);
+            } else {
+                IERC20(marketParams.loanToken).forceApprove(morpho, repayAmount);
+                IMorpho(morpho).repay({
+                    marketParams: marketParams,
+                    assets: repayAmount,
+                    shares: 0,
+                    onBehalf: address(this),
+                    data: new bytes(0)
+                });
+            }
+        }
+
+        if (withdrawAmount > 0) {
+            IMorpho(morpho).withdrawCollateral({
+                marketParams: marketParams,
+                assets: withdrawAmount,
+                onBehalf: address(this),
+                receiver: address(this)
+            });
+        }
+    }
+
+    /**
+     * @notice Supply collateral and borrow assets from a Morpho blue market on behalf of `onBehalf` and send borrowed assets to `receiver`
+     * @param morpho The address of the top level Morpho contract
+     * @param marketParams The market parameters of the individual morpho blue market
+     * @param supplyAssetAmount The amount of assets to supply as collateral
+     * @param borrowAssetAmount The amount of assets to borrow
+     */
+    function supplyCollateralAndBorrow(
+        address morpho,
+        MarketParams memory marketParams,
+        uint256 supplyAssetAmount,
+        uint256 borrowAssetAmount
+    ) external {
+        if (supplyAssetAmount > 0) {
+            IERC20(marketParams.collateralToken).forceApprove(morpho, supplyAssetAmount);
+            IMorpho(morpho).supplyCollateral({
+                marketParams: marketParams,
+                assets: supplyAssetAmount,
+                onBehalf: address(this),
+                data: new bytes(0)
+            });
+        }
+        if (borrowAssetAmount > 0) {
+            IMorpho(morpho).borrow({
+                marketParams: marketParams,
+                assets: borrowAssetAmount,
+                shares: 0,
+                onBehalf: address(this),
+                receiver: address(this)
+            });
+        }
+    }
+
+    // Helper function to convert MarketParams to bytes32 Id
+    // Reference: https://github.com/morpho-org/morpho-blue/blob/731e3f7ed97cf15f8fe00b86e4be5365eb3802ac/src/libraries/MarketParamsLib.sol
+    function marketId(MarketParams memory params) public pure returns (bytes32 marketParamsId) {
+        assembly ("memory-safe") {
+            marketParamsId := keccak256(params, 160)
```

Comment:
why not `keccak256(abi.encode(params))`?

Reply:
this one works too `keccak256(abi.encode(params))`,  actually I was initially using this one happily haha.

But then I saw Morpho's implementation is using `assembly` 
So ended up sticking with their implementation 1:1 as I have put the `reference` comment above to avoid potential confusion in the future (maybe not much).